src/binding.c

```diff
@@ -11,13 +11,47 @@
 #include "pafish/wine.h"
 #include "pafish/sandboxie.h"
 
+
+// these are constant sizes that are guaranteed to not
+// change on a whim.
+#define CPU_VENDOR_STRING_SIZE 13
+#define CPU_BRAND_STRING_SIZE 49
+#define HYPERVISOR_VENDOR_STRING_SIZE 13
+#define HYPERVISOR_MIN_VENDOR_STRING_SIZE 10
+
 #define NAPI_EXPORT_BOOL(name)                                                    \
   {                                                                               \
     napi_value name##_bool;                                                       \
     NAPI_STATUS_THROWS(napi_get_boolean(env, name, &name##_bool))                 \
     NAPI_STATUS_THROWS(napi_set_named_property(env, exports, #name, name##_bool)) \
   }
 
+
+/**
+ * We can't set properties using NAPI_EXPORT_STRING from the napi-macros set since it unconditionally
+ * uses NAPI_STATUS_THROWS_VOID, which means that we potentially corrupt return stack memory due to not
+ * returning an explicit value. Hence, let's create this small function to do that job for us, which is also
+ * nicely usable with the vanilla NAPI_STATUS_THROWS
+ */
+static napi_status set_string(napi_env env, napi_value *object, const char* key, const char* value)
+{
+    napi_value utf8_value;
+    napi_status status = napi_create_string_utf8(env, value, NAPI_AUTO_LENGTH, &utf8_value);
+    if (status != napi_ok)
+    {
+        return status;
+    }
+
+    status = napi_set_named_property(env, *object, key, utf8_value);
+    if (status != napi_ok)
+    {
+        return status;
+    }
+
+
+    return napi_ok;
+}
+
```

Comment:
can't we make a MACRO for this?

Reply:
I strongly dislike using macro's for this type of thing, mostly also because I'm *very bad* at writing those things.

I don't think there will be a real benefit to making this a macro.